vlm/src/main/scala/geotrellis/contrib/vlm/avro/GeoTrellisDataPath.scala

```diff
@@ -37,54 +40,48 @@ import geotrellis.spark.LayerId
  *  @note The order of the query parameters does not matter.
  */
 case class GeoTrellisDataPath(path: String) extends DataPath {
-  private val servicePrefix: String = "gt+"
+  import GeoTrellisDataPath._
 
   private val layerNameParam: String = "layer"
   private val zoomLevelParam: String = "zoom"
   private val bandCountParam: String = "band_count"
 
-  require(path.contains(s"$layerNameParam="), s"The layer query parameter must be in the given path: $path")
-
-  // TODO: Support having the zoom parameter be optional
-  require(path.contains(s"$zoomLevelParam="), s"The zoom query parameter must be in the given path: $path")
-
-  private val strippedPath: String =
-    if (path.startsWith(servicePrefix))
-      path.splitAt(servicePrefix.size)._2
-    else
-      path
-
-  val Array(catalogPath, queryParams) = strippedPath.split('?')
-
-  private val brokenUpParams: Array[(String, String)] =
-    queryParams
-      .split('&')
-      .map {
-        _.split('=') match {
-          case Array(k, v) => (k, v)
-        }
+  // try to parse it, otherwise it is a path
+  private val uri = UrlWithAuthority.parseOption(path).fold(Url().withPath(UrlPath.fromRaw(path)): Url)(identity)
+  private val queryString = uri.query
+
+  lazy val catalogPath: String = {
```

Comment:
Does this `val` need to be `lazy`? I understand that `lazy val`s can give us trouble during run time.

Reply:
I thought about moving the parsing logic our of the `GeoTrellisDataPath` object.